### PR TITLE
Support for non-persistent sessions and table filtering

### PR DESCRIPTION
**Pull request by hegupta-adbe**
Please ensure your pull request adheres to the following guidelines:
- [ ] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues

Thanks for contributing!

### CODE REVIEW

File: src/OneDrive.js [hunk 1]
```diff
@@ -423,7 +423,11 @@ export class OneDrive {
   /**
    */
   getWorkbook(driveItem) {
-    return new Workbook(this, `/drives/${driveItem.parentReference.driveId}/items/${driveItem.id}/workbook`, this.log);
+    return this.getWorkbookFromPath(`/drives/${driveItem.parentReference.driveId}/items/${driveItem.id}/workbook`);
+  }
+
+  getWorkbookFromPath(path) {
```
Comment (dominique-pfister) [inline src/OneDrive.js:429]: what is the use-case of that method?

File: src/excel/Workbook.js [hunk 2]
```diff
@@ -23,13 +23,15 @@ export class Workbook extends NamedItemContainer {
...
-  async createSession() {
+  async createSession(persistent = true) {
     if (this._sessionId) {
       throw new StatusCodeError('This workbook is already associated with a session', 400);
     }
     const uri = `${this.uri}/createSession`;
     const result = await this._oneDrive.doFetch(uri, false, {
       method: 'POST',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify({persistChanges: persistent})
```
Comment (dominique-pfister) [inline src/excel/Workbook.js:34]: wrong linting, space required before `}` and after `{`

File: src/excel/Workbook.js [hunk 3]
```diff
@@ -23,13 +23,15 @@ export class Workbook extends NamedItemContainer {
     this._log = log;
   }
 
-  async createSession() {
+  async createSession(persistent = true) {
```
Comment (dominique-pfister) [inline src/excel/Workbook.js:26]: if you call the parameter `persistChanges`, you can simply write:
```
body: JSON.stringify({ persistChanges })
```
below

File: src/OneDrive.js (hunk 1 above)
Comment (hegupta-adbe) [reply src/OneDrive.js:429]: Githib actions implementation for post-submit processing. Franklin passes us the raw path in the event payload and we'd like to create a workbook object out of that https://github.com/hegupta-adbe/form-data-processor/blob/helix-onedrive/index.js#L202

Comment (dominique-pfister) [reply src/OneDrive.js:429]: I'd rather use the [OneDrive.getDriveItem](https://github.com/adobe/helix-onedrive-support/blob/eff748fa9c1daeacd9c97594241a0e607da0ecae/src/OneDrive.d.ts#L150) API followed by [OneDrive.getWorkbook](https://github.com/adobe/helix-onedrive-support/blob/eff748fa9c1daeacd9c97594241a0e607da0ecae/src/OneDrive.d.ts#L190). Your code change would return a `Workbook` instance with an opaque path that has neither been validated nor tested for existence.

Comment (tripodsan) [reply src/OneDrive.js:429]: or at least: https://github.com/adobe/helix-onedrive-support/blob/eff748fa9c1daeacd9c97594241a0e607da0ecae/src/OneDrive.d.ts#L112C10-L112C26

**Comment by hegupta-adbe** [comment]
@tripodsan @dominique-pfister I've incorporated the comments and fixed all linting errors, and all checks are passing now. Please review and merge if it looks ok now.

**Review by tripodsan — APPROVED** [review]

**Review by dominique-pfister — APPROVED** [review]

**Pull request merged**